specification/keyvault/Security.KeyVault.BackupRestore/models.tsp

```diff
@@ -0,0 +1,244 @@
+import "@typespec/http";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "../Security.KeyVault.Common/common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace KeyVault;
+
+/**
+ * An authentication method and location for the operation.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Model name has already shipped"
+model SASTokenParameter {
+  /**
+   * Azure Blob storage container Uri
+   */
+  storageResourceUri: string;
+
+  /**
+   * The SAS token pointing to an Azure Blob storage container
+   */
+  token?: string;
+
+  /**
+   * Indicates which authentication method should be used. If set to true, Managed HSM will use the configured user-assigned managed identity to authenticate with Azure Storage. Otherwise, a SAS token has to be specified.
+   */
+  useManagedIdentity?: boolean = false;
+}
+
+/**
+ * The status of a long-running operation.
+ */
+@lroStatus
+union OperationStatus {
```

Comment:
why do we have a type not in the swagger?

Reply:
Pasting here what I wrote in our list of TSP-related changes:

> For long running operations, status monitor responses need to [follow a certain format](https://azure.github.io/typespec-azure/docs/howtos/azure-core/long-running-operations/#status-monitor-types-in-azurecore) in TSP. Notably for KV conversion, this means the status of LRO responses now need to [point to a status monitor enum](https://github.com/Azure/azure-rest-api-specs/blob/29226329509494757be9052b932fbbbbe87b1b2c/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.5/backuprestore.json#L326) when the response used to be described as [an arbitrary string](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.5/backuprestore.json#L368). This hypothetically shouldn't be an issue since the status monitor just [describes the possible responses](https://github.com/Azure/azure-rest-api-specs/blob/29226329509494757be9052b932fbbbbe87b1b2c/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.5/backuprestore.json#L370-L405) we could receive from the service, but it's something to keep in mind